tsl/src/nodes/decompress_chunk/exec.c

```diff
@@ -251,8 +256,16 @@ decompress_initialize_batch_state(DecompressChunkState *chunk_state,
 	}
 
 	batch_state->per_batch_context = AllocSetContextCreate(CurrentMemoryContext,
-														   "DecompressChunk per_batch",
-														   ALLOCSET_DEFAULT_SIZES);
+														   "DecompressChunk batch",
+														   /* minContextSize = */ 0,
+														   /* initBlockSize = */ 64 * 1024,
+														   /* maxBlockSize = */ 64 * 1024);
+
+	batch_state->arrow_context = AllocSetContextCreate(CurrentMemoryContext,
```

Comment:
Not every decompression leads to a batch decompression where the `arrow_context` is used. Couldn't we create the memory context on demand?

Reply:
Done.